NullPointerException running liberty:run with SpringBoot jar module

[INFO] CWWKM2185I: The liberty-maven-plugin configuration parameter "appsDirectory" value defaults to "dropins".
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.921 s
[INFO] Finished at: 2020-09-17T21:50:04+08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.openliberty.tools:liberty-maven-plugin:3.2:run (default-cli) on project Dragonheaven: null: MojoExecutionException: NullPointerException -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException

--- a/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/applications/DeployMojo.java
+++ b/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/applications/DeployMojo.java
@@ -195,10 +195,14 @@ public class DeployMojo extends DeployMojoSupport {
     }
 
     private void installSpringBootApp() throws Exception {
+        if (!SpringBootUtil.doesSpringBootRepackageGoalExecutionExist(project)) {
+            throw new MojoExecutionException("The repackage goal of the spring-boot-maven-plugin must be configured to run first in order to create the required executable archive.");
+        }
+
         File fatArchiveSrc = SpringBootUtil.getSpringBootUberJAR(project, getLog());
         
         // Check if the archiveSrc is executable and then invokeSpringUtilCommand. 
-        if(io.openliberty.tools.common.plugins.util.SpringBootUtil.isSpringBootUberJar(fatArchiveSrc)) {
+        if (io.openliberty.tools.common.plugins.util.SpringBootUtil.isSpringBootUberJar(fatArchiveSrc)) {
             File thinArchiveTarget = getThinArchiveTarget(fatArchiveSrc);
             File libIndexCacheTarget = getLibIndexCacheTarget();
             

--- a/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/utils/MavenProjectUtil.java
+++ b/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/utils/MavenProjectUtil.java
@@ -1,5 +1,5 @@
 /**
- * (C) Copyright IBM Corporation 2018.
+ * (C) Copyright IBM Corporation 2018, 2020.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -118,7 +118,32 @@ public class MavenProjectUtil {
         }
         throw new PluginScenarioException("Could not find configuration string " + configName + " for goal " + goal + " on plugin " + pluginKey);
     }
-    
+
+    /**
+     * Checks that the plugin exists for the given pluginKey.
+     *
+     */    
+    public static boolean doesPluginGoalExecutionExist(MavenProject project, String pluginKey, String goal) {
+        boolean exists = false;
+
+        Plugin plugin = project.getPlugin(pluginKey);
+        if (plugin != null) {
+            List<PluginExecution> executions = plugin.getExecutions();
+        
+            if (executions != null) {
+                for(Iterator<PluginExecution> iterator = executions.iterator(); iterator.hasNext();) {
+                    PluginExecution execution = (PluginExecution) iterator.next();
+                    if(execution.getGoals().contains(goal)) {
+                        exists = true;
+                        break;
+                    }
+                }
+            }
+        }
+
+        return exists;
+    }
+
     /**
      * Get an execution of a plugin
      * @param plugin
@@ -148,7 +173,10 @@ public class MavenProjectUtil {
      */
     public static PluginExecution getPluginGoalExecution(MavenProject project, String pluginKey, String goal) throws PluginScenarioException {
         Plugin plugin = project.getPlugin(pluginKey);
-        return getPluginGoalExecution(plugin, goal);
+        if (plugin != null) {
+            return getPluginGoalExecution(plugin, goal);
+        }
+        throw new PluginScenarioException("Could not find plugin " + pluginKey);
     }
     
     /**

--- a/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/utils/SpringBootUtil.java
+++ b/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/utils/SpringBootUtil.java
@@ -1,5 +1,5 @@
 /**
- * (C) Copyright IBM Corporation 2018.
+ * (C) Copyright IBM Corporation 2018, 2020.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -23,6 +23,14 @@ import org.apache.maven.project.MavenProject;
 import io.openliberty.tools.common.plugins.util.PluginScenarioException;
 
 public class SpringBootUtil {
+
+    /**
+     * Checks that the plugin exists for the given pluginKey.
+     *
+     */    
+    public static boolean doesSpringBootRepackageGoalExecutionExist(MavenProject project) {
+        return MavenProjectUtil.doesPluginGoalExecutionExist(project, "org.springframework.boot:spring-boot-maven-plugin", "repackage");
+    }
 
     /**
      * Read the value of the classifier configuration parameter from the

